mm2src/mm2_main/src/rpc/lp_commands/lp_commands_legacy.rs

```diff
@@ -37,45 +37,74 @@ use crate::mm2::lp_ordermatch::{cancel_orders_by, CancelBy};
         Ok(None) => return ERR!("No such coin: {}", ticker),
         Err(err) => return ERR!("!lp_coinfind({}): ", err),
     };
-    let swaps = try_s!(active_swaps_using_coin(&ctx, &ticker));
-    if !swaps.is_empty() {
-        let err = json!({
-            "error": format!("There're active swaps using {}", ticker),
-            "swaps": swaps,
-        });
-        return Response::builder()
-            .status(500)
-            .body(json::to_vec(&err).unwrap())
-            .map_err(|e| ERRL!("{}", e));
-    }
-    let (cancelled, still_matching) = try_s!(cancel_orders_by(&ctx, CancelBy::Coin { ticker: ticker.clone() }).await);
-    if !still_matching.is_empty() {
-        let err = json!({
-            "error": format!("There're currently matching orders using {}", ticker),
-            "orders": {
-                "matching": still_matching,
-                "cancelled": cancelled,
-            }
-        });
-        return Response::builder()
-            .status(500)
-            .body(json::to_vec(&err).unwrap())
-            .map_err(|e| ERRL!("{}", e));
+    let coins_ctx = CoinsContext::from_ctx(&ctx).map_err(|err| ERRL!("{}", err))?;
+
+    // Get all enabled tokens with platform coin including the coin.
+    let coins_to_disable = coins_ctx
+        .get_tokens_to_disable(&ticker)
+        .await
+        .into_iter()
+        .chain(std::iter::once(coin.platform_ticker().to_string()));
+
+    let mut disabled_tokens_tickers = vec![];
+    let mut cancelled_orders = vec![];
+    for ticker in coins_to_disable {
+        log!("disabling {ticker} coin");
+        let swaps = try_s!(active_swaps_using_coin(&ctx, &ticker));
+        if !swaps.is_empty() {
+            let err = json!({
+                "error": format!("There're active swaps using {}", &ticker),
+                "swaps": swaps,
+            });
+            return Response::builder()
+                .status(INTERNAL_SERVER_ERROR_CODE)
+                .body(json::to_vec(&err).unwrap())
+                .map_err(|e| ERRL!("{}", e));
+        }
+        let (cancelled, still_matching) = try_s!(
+            cancel_orders_by(&ctx, CancelBy::Coin {
+                ticker: ticker.to_string()
+            })
+            .await
+        );
+
+        if !still_matching.is_empty() {
+            let err = json!({
+                "error": format!("There're currently matching orders using {}", ticker),
+                "orders": {
+                    "matching": still_matching,
+                    "cancelled": &cancelled,
+                }
+            });
+            return Response::builder()
+                .status(INTERNAL_SERVER_ERROR_CODE)
+                .body(json::to_vec(&err).unwrap())
+                .map_err(|e| ERRL!("{}", e));
+        }
+
+        try_s!(coins_ctx.remove_coin(&ctx, &ticker).await);
+
+        // Combine all orders to a single vector
+        cancelled_orders.extend(cancelled);
+        disabled_tokens_tickers.push(ticker);
```

Comment:
Returning errors from inside the for loop when some tokens have been disabled is not a good approach. We can either return the list of disabled tokens with the error, or check for active swaps/currently matching orders first before disabling any tokens (I think this second approach is better).

Reply:
good note! I think the both options will be used! thank you